lib/index.js

```diff
@@ -1,346 +1,61 @@
-   *
-   * @protected
-   */
-  getObjectFromBody(
-    request,
-    property,
-    subPropertyTypeMapping = null,
-    defaultValue
-  ) {
-    if (
-      (!request.input.body || !request.input.body[property]) &&
-      typeof defaultValue === 'undefined'
-    ) {
-      this.throwError('BadRequest', `Missing body property "${property}"`);
-    }
-
-    if (
-      request.input.body[property] &&
-      typeof request.input.body[property] !== 'object'
-    ) {
-      this.throwError(
-        'BadRequest',
-        `Invalid body property "${property} value "${
-          request.input.body[property]
-        }"`
-      );
-    }
-
-    const value = request.input.body[property]
-      ? request.input.body[property]
-      : defaultValue;
-
-    if (subPropertyTypeMapping === null) {
-      return value;
-    }
-
-    if (
-      !Object.keys(value).every(field =>
-        Object.keys(subPropertyTypeMapping).includes(field)
-      ) ||
-      !Object.keys(subPropertyTypeMapping).every(
-        field => typeof value[field] === subPropertyTypeMapping[field]
-      )
-    ) {
-      this.throwError('BadRequest', `Bad sub-property in property ${property}`);
-    }
-
-    return value;
-  }
-
-  /**
-   * @param {Error} error
-   */
-  outputError(error) {
-    this.context.log.error(error.message).catch(() => {});
-    this.context.log.error(error.stack).catch(() => {});
-  }
-
-  /**
-   * @param {string} errorType
-   * @param {string|null} [message]
-   * @throws {KuzzleError}
-   *
-   * @protected
-   */
-  throwError(errorType, message) {
-    throw new Error(message);
-  }
-
-  /**
-   * An example of a controller function. It is called by the controller/action
-   * routes defined in the `controllers` object above. It takes the request as
-   * an argument and must return a Promise.
-   *
-   * @param {Request} request The request sent to the controller/action route
-   * @return {Promise} A promise resolving the response of the route.
-   *
-   * @see https://docs.kuzzle.io/plugins-reference/plugins-features/adding-controllers/
-   * @see https://docs.kuzzle.io/guide/essentials/request-and-response-format/
-   */
   async graphQLEndpoint(request) {
-    const query = this.getStringFromBody(request, 'query');
-    const variables = this.getObjectFromBody(request, 'variables', null, null);
-
-    const graphqlResult = await graphql(this.graphQLSchema, query, variables);
-
-    if (graphqlResult.errors) {
-      graphqlResult.errors.forEach(error => this.outputError(error));
-    }
-
-    const result = Buffer.from(JSON.stringify(graphqlResult));
-
-    request.setResult(result, {
-      raw: true,
-      headers: {
-        'Content-Type': 'application/json'
-      }
-    });
+    const body = request.input.body || {};
+    const query = body.query || {};
+    const vars = body.variables || {};
+    const graphqlResult = await graphql(this.graphQLSchema, query, vars);
 
-    return result;
+    return JSON.stringify(graphqlResult);
```

Comment:
You don't need to stringify the result yourself, Kuzzle protocols will automatically convert the return of a controller action to the appropriate format (eg [http protocol](https://github.com/kuzzleio/kuzzle/blob/master/lib/core/entrypoints/protocols/http.js#L321)